src/overlays/actors/ovl_En_Kusa/z_en_kusa.c

```diff
@@ -7,72 +16,485 @@
+    s32 pad; // padding to push the stack down
+    Vec3f contactPos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B89C.s")
+    if (this->actor.bgCheckFlags & 11) {
+        if ((this->actor.bgCheckFlags & 32) == 0) {
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot, 20, NA_SE_EV_PLANT_BROKEN);
+        }
+        func_80A9B21C(this, globalCtx);
+        func_80A9B07C(this, globalCtx); // Drop collectible
+        switch (this->actor.params & 3) {
+            case 0:
+            case 2:
+                Actor_Kill(&this->actor);
+                return;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9B8D8.s")
+            case 1:
+                func_80A9BF3C(&this->actor);
+        }
+    } else {
+        if (this->actor.bgCheckFlags & 0x40) {
+            contactPos.x = this->actor.posRot.pos.x;
+            contactPos.y = this->actor.posRot.pos.y + this->actor.waterY;
+            contactPos.z = this->actor.posRot.pos.z;
+            EffectSsGSplash_Spawn(globalCtx, &contactPos, NULL, NULL, 0, 400);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 150, 650, 0);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 400, 800, 4);
+            EffectSsGRipple_Spawn(globalCtx, &contactPos, 500, 1100, 8);
+            this->actor.minVelocityY = -3.0f;
+            *D_80A9C1D4 = (*D_80A9C1D4 >> 1);
+            *D_80A9C1D0 = (*D_80A9C1D0 >> 1);
+            *D_80A9C1DC = (*D_80A9C1DC >> 1);
+            *D_80A9C1D8 = (*D_80A9C1D8 >> 1);
+            this->actor.bgCheckFlags = this->actor.bgCheckFlags & 0xFFBF;
+            Audio_PlaySoundAtPosition(globalCtx, &this->actor.posRot, 40, NA_SE_EV_DIVE_INTO_WATER_L);
+        }
+        func_80A9B140(&this->actor);
+        Math_ApproxS(D_80A9C1D4, *D_80A9C1D0, 0x1F4);
+        Math_ApproxS(D_80A9C1DC, *D_80A9C1D8, 0xAA);
+        this->actor.shape.rot.x += *D_80A9C1D4;
+        this->actor.shape.rot.y += *D_80A9C1DC;
+        func_80A9B174(&this->actor.velocity, 0.05f);
+        func_8002D7EC(&this->actor);
+        func_8002E4B4(globalCtx, this, 7.5f, 35.0f, 0.0f, 0xC5);
+        Collider_CylinderUpdate(&this->actor, &this->collider);
+        CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BA98.s")
+void func_80A9BEAC(EnKusa* this) {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BAD8.s")
+    if ((this->actor.params & 3) != 1) {
+        if ((this->actor.params & 3) == 2) {
+            // This function is blank
+            func_80A9AFA0(this, func_80A9BF30);
+        }
+    } else {
+        func_80A9AFA0(this, func_80A9BEFC);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Kusa/func_80A9BBB0.s")
+void func_80A9BEFC(EnKusa* this, GlobalContext* globalCtx) {
+    if (this->unk_19C >= 0x78) {
```

Comment:
```suggestion
    if (this->unk_19C >= 120) {
```
this->unk_19C is probably some sort of timer, so it should compare to decimal numbers